src/components/Button/Button.types.ts

```diff
@@ -81,6 +81,10 @@ export type Props = IrisProps<
      */
     textShift?: boolean;
     theme?: any;
+    /**
+     * HTML Button type property
+     */
+    type?: 'button' | 'reset' | 'submit';
```

Comment:
I was going to suggest you type this directly to the `HTMLButtonInterface` type key value, but was surprised to see it's just `string`: https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L6163

Are these the only 3 valid strings for button `type`? If so, it could be a could idea to make this a separate exported type `ButtonType` that can be used throughout Iris, as well as document what each value means in the JSDoc comment.

Reply:
According to MDN there are only 3 possible values https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type. I guess they went with just string to be extensible just in case a new type pops up.